### ISSUE TITLE
[FEATURE] Install Gradle Wrapper Validation Action org-wide

### ISSUE THREAD

**Issue by dbwiddis** [opensearch-project/.github#116]
### Is your feature request related to a problem?

The Gradle Wrapper jar file is used extensively by organization repos to run tests.  The jar file included with the installation is a binary executable.  A nefarious actor could attempt to upload an arbitrary executable containing malicious code under an innocent-sounding PR like "Update Gradle Wrapper".  I actually did such a PR (although with valid code) [here](https://github.com/opensearch-project/opensearch-build/pull/2289) and [here](https://github.com/opensearch-project/opensearch-sdk-java/pull/15).

Here's what the updated binary looked like during review:

![Screen Shot 2022-12-08 at 9 09 19 AM](https://user-images.githubusercontent.com/9291703/206518049-83fd89a4-bfde-4774-a7a4-b0c9c7eff76f.png)

I'm not sure if the reviewers validated that it was legitimate or just trusted me. :-)

Gradle publishes a [Gradle Wrapper Validation Action](https://github.com/gradle/wrapper-validation-action) which can easily be added to a repo's PR workflow. All it does is validate that the SHA-256 checksum matches a legitimate release.

That validation workflow is only presently installed on [three opensearch-project repos](https://github.com/search?q=org%3Aopensearch-project+wrapper-validation-action&type=code).

### What solution would you like?

Install that workflow to run organization-wide.

### What alternatives have you considered?

Recommending all repo-owners create the workflow, duplicating the configuration.

### Do you have any additional context?

See the [Gradle Wrapper Validation Action](https://github.com/gradle/wrapper-validation-action) README.

**Comment by dblock** [comment]
Checksum validation at runtiime is a good idea, let's do it. But I think the "malicious actor" story doesn't quite apply, because you're not such an actor and there's a code review from another actor. It's just that we can't examine binaries, so trust but verify.

**Comment by saratvemulapalli** [comment]
@dbwiddis unfortunately we have other repos which dont use java/gradle build system.

**Comment by dblock** [comment]
@dbwiddis What do you want to do about this one?

**Comment by dbwiddis** [comment]
I plan to convert this to a template: https://docs.github.com/en/actions/using-workflows/creating-starter-workflows-for-your-organization

Just haven't had the time to do it but haven't forgotten.

**Comment by dbwiddis** [comment]
I've updated my PR #118 to create a template.  It will show as a [Starter workflow](https://docs.github.com/en/actions/using-workflows/using-starter-workflows) for any project that has a gradle wrapper in its project root.

LMK if you want me to run a campaign to suggest maintainers who use the gradle wrapper to use this workflow.

Once the PR is merged I will create a separate PR to add instructions to https://github.com/opensearch-project/opensearch-plugins/blob/main/WORKFLOWS.md

**Comment by dbwiddis** [comment]
Closed #118. Leaving this issue open as a reminder to me to add the workflow in `opensearch-plugins`.  Maintainers feel free to move this issue there.
